File: ReadActivity.java
package com.example.fitpeak;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.ProgressBar;
import android.widget.RelativeLayout;
import android.widget.Toast;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.android.volley.Request;
import com.android.volley.Response;
import com.android.volley.toolbox.JsonObjectRequest;
import com.android.volley.toolbox.Volley;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;
import java.util.ArrayList;

public class ReadActivity extends AppCompatActivity {

    private RelativeLayout readProgressLayout;
    private ProgressBar readProgressBar;
    private RecyclerView recyclerView;
    private LinearLayoutManager layoutManager;
    private ReadRecyclerAdapter recyclerAdapter;
    private ArrayList<Book> fitList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_read);

        readProgressLayout=findViewById(R.id.readProgressLayout);
        readProgressBar=findViewById(R.id.readProgressBar);
        recyclerView=findViewById(R.id.recyclerView);
        layoutManager= new LinearLayoutManager(this);

        final JsonObjectRequest jsonObjectRequest = new JsonObjectRequest(Request.Method.GET,
                "https://script.google.com/macros/s/AKfycbyXe3cAdQ6tmIEvH9MKRxGyY-yuuWzCMYsKkZyK9qLb_0_STbWaJOqqbZx5oa9E1p6q/exec",
                null,
                new Response.Listener<JSONObject>() {
                    @Override
                    public void onResponse(JSONObject response) {
                        readProgressLayout.setVisibility(View.GONE);
                        readProgressBar.setVisibility(View.GONE);

                        try {
                            JSONArray data = response.getJSONArray("fitList");
                            for(int i=0; i<data.length(); i++){
                                JSONObject bookJsonObject = data.getJSONObject(i);
                                Book bookObject = new Book(
                                        bookJsonObject.getString("personName"),
                                        bookJsonObject.getString("personActivity"),
                                        bookJsonObject.getInt("activityData")
                                );
                                fitList.add(bookObject);
                            }
                            recyclerAdapter= new ReadRecyclerAdapter(ReadActivity.this,fitList);
                            recyclerView.setAdapter(recyclerAdapter);
                            recyclerView.setLayoutManager(layoutManager);

                        } catch (JSONException e) {
                            e.printStackTrace();
                            Toast.makeText(ReadActivity.this, e.toString(), Toast.LENGTH_SHORT).show();
                        }

                    }
                },
                new Response.ErrorListener() {
                    @Override
                    public void onErrorResponse(com.android.volley.VolleyError error) {
                        readProgressLayout.setVisibility(View.GONE);
                        readProgressBar.setVisibility(View.GONE);
                        Toast.makeText(ReadActivity.this, error.toString(), Toast.LENGTH_SHORT).show();
                    }
                });

        Volley.newRequestQueue(this).add(jsonObjectRequest);

    }
}

File: ReadActivity.kt
package com.pritampattanayak.tutorial

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.ProgressBar
import android.widget.RelativeLayout
import android.widget.Toast
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.android.volley.Request
import com.android.volley.Response
import com.android.volley.toolbox.JsonObjectRequest
import com.android.volley.toolbox.Volley

class ReadActivity : AppCompatActivity() {

    lateinit var readProgressLayout:RelativeLayout
    lateinit var readProgressBar:ProgressBar
    lateinit var recyclerView:RecyclerView
    lateinit var layoutManager: LinearLayoutManager
    lateinit var recyclerAdapter: ReadRecyclerAdapter
    

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_read)

        readProgressLayout=findViewById(R.id.readProgressLayout)
        readProgressBar=findViewById(R.id.readProgressBar)
        recyclerView=findViewById(R.id.recyclerView)
        layoutManager= LinearLayoutManager(this)

        val bookList= arrayListOf<Book>()

        val queue=Volley.newRequestQueue(this)
        val url="https://script.google.com/macros/s/AKfycbxwop4CsnSVjGjkrtBeFXrT3uQZbyUEO0cR-AVVp545_yOzSV2xTsbx98Na37UlBavn/exec"
        val jsonObjectRequest=object :JsonObjectRequest(
            Request.Method.GET,url,null,
            Response.Listener {
                readProgressLayout.visibility=View.GONE
                readProgressBar.visibility=View.GONE
                val data=it.getJSONArray("bookList")
                for(i in 0 until data.length()){
                    val bookJasonObject=data.getJSONObject(i)
                    val bookObject=Book(
                        bookJasonObject.getString("itemName"),
                        bookJasonObject.getString("itemAuthor"),
                        bookJasonObject.getInt("itemPrice"),
                        bookJasonObject.getString("itemRating")
                    )
                    bookList.add(bookObject)
                }
                recyclerAdapter= ReadRecyclerAdapter(this,bookList)
                recyclerView.adapter=recyclerAdapter
                recyclerView.layoutManager=layoutManager
            },Response.ErrorListener {
                readProgressLayout.visibility=View.GONE
                readProgressBar.visibility=View.GONE
                Toast.makeText(this@ReadActivity, it.toString(), Toast.LENGTH_SHORT).show()
            }
        ){
            override fun getHeaders(): MutableMap<String, String> {
                return super.getHeaders()
            }
        }
        queue.add(jsonObjectRequest)
    }
}